packages/jest-cli/src/TestScheduler.js

```diff
@@ -85,25 +84,16 @@ export default class TestScheduler {
       getEstimatedTime(timings, this._globalConfig.maxWorkers) / 1000,
     );
 
-    // Run in band if we only have one test or one worker available, unless we
-    // are using the watch mode, in which case the TTY has to be responsive and
-    // we cannot schedule anything in the main thread. Same logic applies to
-    // watchAll.
-    //
-    // If we are confident from previous runs that the tests will finish
-    // quickly we also run in band to reduce the overhead of spawning workers.
-    const areFastTests = timings.every(timing => timing < SLOW_TEST_TIME);
-
-    const runInBandWatch = tests.length <= 1 && areFastTests;
-    const runInBandNonWatch =
-      this._globalConfig.maxWorkers <= 1 ||
-      tests.length <= 1 ||
-      (tests.length <= 20 && timings.length > 0 && areFastTests);
-
-    const runInBand =
-      this._globalConfig.watch || this._globalConfig.watchAll
-        ? runInBandWatch
-        : runInBandNonWatch;
+    // if runInBand set by config we do not try to compute the value
+    let runInBand = this._globalConfig.runInBand;
```

Comment:
Oh, _now_ I remember. We do this on purpose, because it's easy to leave `runInBand` in the config and make it look like Jest is slow all the time. See: https://github.com/facebook/jest/issues/3215

So, it's only supported through CLI args on purpose:
https://github.com/facebook/jest/blob/700e0dadb85f5dc8ff5dac6c7e98956690049734/packages/jest-config/src/getMaxWorkers.js#L14-L17

The correct fix would be to add `this._globalConfig.maxWorkers <= 1` condition to `runInBandWatch`, can you make that work?

Reply:
ok i see! this will fix the` watchAll`  regression also. Sure i'll do the change and adjust tests

Reply:
Thank you!